modules/hdf5_handler/h5dmr.cc

```diff
@@ -628,11 +635,70 @@ read_objects_base_type(D4Group * d4_grp, hid_t pid, const string & varname, cons
 
         dimnames_size = (int) (dt_inst.dimnames.size());
 
+        // Here we need to add the unlimited dimension info if dimension scale dimension names are present.
```

Comment:
This might be hard to do, but could this part of the function be extracted to its own function? That would simplify the code. If it's too hard to do that, this can stay as it is. Your call on that.